client-java/controller/src/main/java/org/evomaster/client/java/controller/problem/rpc/schema/params/ByteBufferParam.java

```diff
@@ -0,0 +1,98 @@
+package org.evomaster.client.java.controller.problem.rpc.schema.params;
+
+import org.evomaster.client.java.controller.api.dto.problem.rpc.ParamDto;
+import org.evomaster.client.java.controller.problem.rpc.CodeJavaGenerator;
+import org.evomaster.client.java.controller.problem.rpc.schema.types.AccessibleSchema;
+import org.evomaster.client.java.controller.problem.rpc.schema.types.ByteBufferType;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * this is created for handling binary in thrift, see https://thrift.apache.org/docs/types
+ * handle it as string
+ */
+public class ByteBufferParam extends NamedTypedValue<ByteBufferType, ByteBuffer>{
+
+    public ByteBufferParam(String name, AccessibleSchema accessibleSchema) {
+        super(name, new ByteBufferType(), accessibleSchema);
+    }
+
+    public void setValue(byte[] value) {
+        ByteBuffer buffer = ByteBuffer.allocate(value.length);
+        buffer.put(value);
+        this.setValue(buffer);
+    }
+
+    @Override
+    public Object newInstance() throws ClassNotFoundException {
+        return getValue();
+    }
+
+    @Override
+    public ParamDto getDto() {
+        ParamDto dto = super.getDto();
+        if (getValue() != null){
+            // bytebuffer is now handled as string
+            dto.stringValue = new String(getValue().array(), StandardCharsets.UTF_8);
+        }
+
+        return dto;
+    }
+
+    @Override
+    public ByteBufferParam copyStructure() {
+        return new ByteBufferParam(getName(), accessibleSchema);
+    }
+
+    @Override
+    public void setValueBasedOnDto(ParamDto dto) {
+        if (dto.stringValue != null)
+            setValue(dto.stringValue.getBytes());
+    }
+
+    @Override
+    protected void setValueBasedOnValidInstance(Object instance) {
+        setValue((ByteBuffer) instance);
+    }
+
+    @Override
+    public List<String> newInstanceWithJava(boolean isDeclaration, boolean doesIncludeName, String variableName, int indent) {
```

Comment:
so, this is about the discussion we had, in which currently the test case is generated on the Driver side instead of the Core?